src/ipc/ipc.c

```diff
@@ -455,28 +455,20 @@ int ipc_init(struct sof *sof)
 	trace_ipc("ipc_init()");
 
 	/* init ipc data */
-	sof->ipc = rzalloc(SOF_MEM_ZONE_SYS, 0, SOF_MEM_CAPS_RAM,
-			   sizeof(*sof->ipc));
-	sof->ipc->comp_data = rzalloc(SOF_MEM_ZONE_SYS, 0, SOF_MEM_CAPS_RAM,
-				      SOF_IPC_MSG_MAX_SIZE);
+	sof->ipc = rzalloc(SOF_MEM_ZONE_SYS, SOF_MEM_FLAG_SHARED,
```

Comment:
I assume ipc_init() is only called on the master core?

Reply:
@paulstelian97 Yes.